### PR TITLE
vsphere: pass vcpu and ram to cloneSpec

### PR DESCRIPTION
**Pull request by orymate**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?
VCPU and RAM inputs were missing from the clone specification when creating nodes.

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)

### CODE REVIEW

File: internal/providers/vsphere/pke/workflow/create_node_activity.go [hunk 1]
```diff
@@ -122,7 +122,11 @@ func generateVMConfigs(input CreateNodeActivityInput) (*types.VirtualMachineConf
 		return nil, err
 	}
 
-	vmConfig := types.VirtualMachineConfigSpec{}
+	vmConfig := types.VirtualMachineConfigSpec{
+		NumCPUs:           int32(input.VCPU),
+		NumCoresPerSocket: int32(input.VCPU),
```
Comment (pregnor) [inline internal/providers/vsphere/pke/workflow/create_node_activity.go:127]: **Note**: I'm kinda out of the loop, but do we have 1 socket only (whatever it refers to)?

Before starting to look into what these config values mean exactly my intuition said `NumCPUs` is probably the total value and `socket * NumCoresPerSocket` should not be over that, but I could be completely wrong, I'm going to search for these, just wanted to start the discussion in the meantime.

**Review by pregnor — APPROVED** [review]
As far as I could tell with my limited contextual knowledge it looked good.

File: internal/providers/vsphere/pke/workflow/create_node_activity.go (hunk 1 above)
Comment (orymate) [reply internal/providers/vsphere/pke/workflow/create_node_activity.go:127]: I think there's no major drawback of putting all the virtual cores into one virtual socket (and I think the above spec means exactly that), but some strange commercial license checks tend to prefer this layout, and that's why I normally opt for it.

Comment (orymate) [reply internal/providers/vsphere/pke/workflow/create_node_activity.go:127]: Additionally, if you want to have a custom layout, you can always set the vcpu value to zero, and that will fall back to the template's settings.

Comment (pregnor) [reply internal/providers/vsphere/pke/workflow/create_node_activity.go:127]: Thanks.

**Review by sancyx — APPROVED** [review]

**Pull request merged**